### PR TITLE
Bugfix - Allow configuration of disableChallengeResourceVerification property of AKV SecretClient

### PR DESCRIPTION
**Pull request by nagyesta**
# Description

Fixes https://github.com/Azure/azure-sdk-for-java/issues/36561
- Adds challengeResourceVerificationEnabled property to properties objects
- Includes new property in mapping methods
- Configures SecretClient in Factory when challengeResourceVerificationEnabled is set
- Configures CertificateClient in Factory when challengeResourceVerificationEnabled is set
- Updates/adds new tests
- Updates Changelog

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by nagyesta** [comment]
@microsoft-github-policy-service agree

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/common/AzureKeyVaultProperties.java [hunk 1]
```diff
@@ -19,6 +19,8 @@ public class AzureKeyVaultProperties extends AbstractAzureHttpConfigurationPrope
      */
     private String endpoint;
 
+    private boolean disableChallengeResourceVerification = false;
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/common/AzureKeyVaultProperties.java:22]: We have a configuration property named `spring.cloud.azure.credential.managed-identity-enabled`, so I am wondering whether we should call this one `challengeResourceVerificationEnabled` and give it a default value `true`.

Spring Boot configuration annotation processor will read the Java Doc on the field to generate the configuration metadata, https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html. So we should add the description to the field as well.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by saragluna** [comment]
@nagyesta thanks for the PR, this PR looks great, I left some comments to your PR, please check.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/common/AzureKeyVaultProperties.java (hunk 1 above)
Comment (nagyesta) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/common/AzureKeyVaultProperties.java:22]: Hi,
works for me, thank you! I will update the PR tonight with the renamed property names (both in the PropertySource specific  and the Secret/Certificate specific Properties objects) and will make sure to add JavaDoc to the fields as well!

Comment (nagyesta) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/common/AzureKeyVaultProperties.java]: Done, thank you!

**Comment by nagyesta** [comment]
Hi @saragluna , 
I have completed the requested changes, please let me know in case anything else is needed. 
Thank you!

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/secrets/properties/AzureKeyVaultPropertySourceProperties.java [hunk 2]
```diff
@@ -138,4 +144,21 @@ public Duration getRefreshInterval() {
     public void setRefreshInterval(Duration refreshInterval) {
         this.refreshInterval = refreshInterval;
     }
+
+    /**
+     *
+     * @return Whether we should disable the challenge resource verification
```
Comment (nagyesta) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/keyvault/secrets/properties/AzureKeyVaultPropertySourceProperties.java:150]: Missed this, sorry! Let me fix it...

**Comment by saragluna** [comment]
@nagyesta I changed the base branch to `feature/spring-boot-3`, which is our base branch for Spring Boot 3 development.

**Comment by saragluna** [comment]
@nagyesta I pushed one [commit](https://github.com/Azure/azure-sdk-for-java/pull/36603/commits/48bd6bd92ace5d3217928f6e311ae903db6a420f) to your PR, the changes including:
- Refine the javadoc description, according to the Spring Boot doc: https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.developing-auto-configuration.custom-starter.configuration-keys (check the "Here are some rules we follow internally to make sure descriptions are consistent:" part)
- Add one missing configuration metadata in the `additional-spring-configuration-metadata.json` file. You could build the project locally, and a file `target/classes/META-INF/spring-configuration-metadata.json` will be generated under the autoconfigure module. This entry is missing because the Spring Boot annotation processor can't handle the nested collection objects correctly.

File: sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/certificates/CertificateClientBuilderFactory.java [hunk 3]
```diff
@@ -88,6 +88,12 @@ protected void configureService(CertificateClientBuilder builder) {
         PropertyMapper map = new PropertyMapper();
         map.from(certificateClientProperties.getEndpoint()).to(builder::vaultUrl);
         map.from(certificateClientProperties.getServiceVersion()).to(builder::serviceVersion);
+        map.from(certificateClientProperties.isChallengeResourceVerificationEnabled()).to(
+            enabled -> {
+                if (!enabled) {
+                    builder.disableChallengeResourceVerification();
+                }
+            });
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/certificates/CertificateClientBuilderFactory.java:96]: Can change to 
```java
map.from(secretClientProperties.isChallengeResourceVerificationEnabled()).whenFalse().to(t -> builder.disableChallengeResourceVerification());
```

File: sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/secrets/SecretClientBuilderFactory.java [hunk 4]
```diff
@@ -88,6 +88,12 @@ protected void configureService(SecretClientBuilder builder) {
         PropertyMapper map = new PropertyMapper();
         map.from(secretClientProperties.getEndpoint()).to(builder::vaultUrl);
         map.from(secretClientProperties.getServiceVersion()).to(builder::serviceVersion);
+        map.from(secretClientProperties.isChallengeResourceVerificationEnabled()).to(
+            enabled -> {
+                if (!enabled) {
+                    builder.disableChallengeResourceVerification();
+                }
+            });
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/secrets/SecretClientBuilderFactory.java:96]: Same here.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by saragluna** [comment]
@nagyesta I left two more comments.

File: sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/certificates/CertificateClientBuilderFactory.java (hunk 3 above)
Comment (nagyesta) [reply sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/certificates/CertificateClientBuilderFactory.java]: Pushed a fix, thank you!

File: sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/secrets/SecretClientBuilderFactory.java (hunk 4 above)
Comment (nagyesta) [reply sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/keyvault/secrets/SecretClientBuilderFactory.java]: Pushed a fix, thank you!

**Comment by nagyesta** [comment]
Hi @saragluna ,
thank you for making the changes you have mentioned! I have updated the branch with the requested simplifications in the factory calls. Please let me know in case further changes are required!

Regarding the base branch, I wasn't aware of it sorry!  Will it be a problem that the PR now contains the version bump specific changes from the release branch?

**Comment by saragluna** [comment]
Thanks @nagyesta, no worries, the version bump is already in the base branch, so it won't be an issue.

**Review by saragluna — APPROVED** [review]

**Comment by saragluna** [comment]
@Netyyyy, let's cherry pick this feature to the main branch.

**Pull request merged**